### PR TITLE
Result: make them real plugins [v0]

### PR DESCRIPTION
**Pull request by clebergnu**
This is an early preview (v0) of the work to move result out of the job context, and while doing that, making them real plugins.

### CODE REVIEW

**Comment by clebergnu** [comment]
@apahim your review and general test feedback is appreciated!

File: selftests/unit/test_xunit.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def test(self):
         self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
         args = argparse.Namespace()
         args.xunit_output = self.tmpfile[1]
-        self.test_result = xunit.xUnitTestResult(FakeJob(args))
+        self.test_result = xunit.xUnitResult(FakeJob(args))
```
Comment (apahim) [inline selftests/unit/test_xunit.py]: Ok

File: avocado/core/xunit.py [hunk 2]
```diff
@@ -211,7 +211,7 @@ def end_tests(self):
                   'errors': self.errors + self.interrupted,
                   'failures': self.failed,
                   'skip': self.skipped,
-                  'total_time': self.total_time}
+                  'tests_total_time': self.tests_total_time}
```
Comment (apahim) [inline avocado/core/xunit.py]: Ok

File: avocado/core/result.py [hunk 3]
```diff
@@ -165,6 +166,7 @@ def end_test(self, state):
         """
         self.tests_run += 1
         self.tests_total_time += state.get('time_elapsed', -1)
+        self.tests.append(state)
```
Comment (apahim) [inline avocado/core/result.py]: ok

File: avocado/plugins/run.py [hunk 4]
```diff
@@ -185,4 +186,13 @@ def run(self, args):
...
+        job_run = job_instance.run()
+        result_dispatcher = ResultDispatcher()
+        if result_dispatcher.extensions:
+            # At this point job_instance doesn't have a single results attribute
+            # which is the end goal.  For now, we pick any of the plugin classes
+            # added to the result proxy.
+            if len(job_instance.result_proxy.output_plugins) > 0:
+                result = job_instance.result_proxy.output_plugins[0]
+                result_dispatcher.map_method('render', result, job_instance)
+        return job_run
```
Comment (apahim) [inline avocado/plugins/run.py]: ok

**Comment by apahim** [comment]
@clebergnu the code is pretty straight forward and breaks nothing that I could find. Since this is the early preview, I'm looking forward to see the final version.

File: avocado/core/result.py [hunk 5]
```diff
@@ -165,6 +166,7 @@ def end_test(self, state):
         """
         self.tests_run += 1
```
Comment (ldoktor) [inline avocado/core/result.py]: as you're storing the full state, you should remove the `self.tests_run` as the information is duplicate.

File: avocado/plugins/run.py [hunk 6]
```diff
@@ -185,4 +186,13 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
-        return job_instance.run()
+        job_run = job_instance.run()
+        result_dispatcher = ResultDispatcher()
+        if result_dispatcher.extensions:
+            # At this point job_instance doesn't have a single results attribute
```
Comment (ldoktor) [inline avocado/plugins/run.py]: OK, so the single results attribute means that currently results (test statues) are stored per each output_plugin in `results_proxy` and the target goal is to remove the `result_proxy` and store the overall results directly in `job_instance`, right?

Comment (clebergnu) [inline avocado/plugins/run.py]: Right.

File: avocado/core/result.py (hunk 5 above)
Comment (clebergnu) [inline avocado/core/result.py]: This is indeed duplicate information, but public.  It's used, for example, in the TAP plugin.  The `Result` class will receive major changes and cleanups, so my idea is to leave that kind of change to the next stage.  How does it sound?

**Comment by clebergnu** [comment]
Closing in favor of #1352

File: avocado/core/result.py (hunk 5 above)
Comment (ldoktor) [inline avocado/core/result.py]: OK, could you please add `#FIXME:` or `#TODO` mark that it is going to be removed?

**Pull request closed without merging**